### PR TITLE
Add Roster and deploy Roster data to qml

### PR DESCRIPTION
**Pull request by geobra**
This code requests the roster from the server of the logged in jid user. Roster data gets extracted from server and a qt model will be created. This can be displayed in a qml ui. The qml file is just a POC. Feel free to exchange this with real UI.

### CODE REVIEW

File: app/kaidan.pro [hunk 1]
```diff
@@ -17,12 +17,16 @@ LIBS += -L$${SWIFT3PATH}/Swiften -L$${SWIFT3PATH}/3rdParty/Boost $${SWIFTLIB}
 DEFINES += BOOST_SIGNALS_NO_DEPRECATION_WARNING
 
 SOURCES += source/main.cpp \
-        source/EchoBot.cpp
+        source/Kaidan.cpp \
+    source/RosterContoller.cpp \
+    source/RosterItem.cpp
```
Comment (LNJ2) [inline app/kaidan.pro]: Please keep this uniformly. (Use 8 blanks OR 4)

File: app/kaidan.pro [hunk 2]
```diff
@@ -17,12 +17,16 @@ LIBS += -L$${SWIFT3PATH}/Swiften -L$${SWIFT3PATH}/3rdParty/Boost $${SWIFTLIB}
...
 
-HEADERS += source/EchoBot.h \
-            sourc/EchoPayload.h \
+HEADERS += source/Kaidan.h \
+            source/EchoPayload.h \
             source/EchoPayloadParserFactory.h \
-            source/EchoPayloadSerializer.h
+            source/EchoPayloadSerializer.h \
+    source/RosterContoller.h \
+    source/RosterItem.h
```
Comment (LNJ2) [inline app/kaidan.pro]: Same as above; Use 8 or 4 blanks, but use them everywhere.

File: app/source/Kaidan.h [hunk 3]
```diff
@@ -1,33 +1,43 @@
 #ifndef ECHOBOT_H
 #define ECHPBOT_H
```
Comment (LNJ2) [inline app/source/Kaidan.h]: Replace `ECHOBOT_H` with `KAIDAN_H`

**Comment by LNJ2** [comment]
Ah, and please resolve the merge conflict.

**Comment by MRZA-MRZA** [comment]
I have trouble compiling and running it.
What is the reason behind changing SWIFT3PATH? It won't build in the default configuration.
And in app/source/main.cpp:29 view.setSource(QUrl::fromLocalFile("main.qml"));
main.qml does not exit in the runtime. I assume that problem is on my side but I'd like to know what is wrong.

**Comment by geobra** [comment]
The SWIFT3PATH variable is not an optimal solution. Later we should find an other way to solve this. For now, one has to set this correct to reflect his local setup. The qml is just a POC. To get it run, the qml file has to lay side by side with the executable. Be aware of that when you do shadow builds!

File: app/resources/main.qml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import QtQuick 2.1
-import QtQuick.Window 2.0
+import harbour.kaidan 1.0
```
Comment (JBBgameich) [inline app/resources/main.qml]: why **harbour**.kaidan? Just wanted to ask ...

Comment (geobra) [inline app/resources/main.qml]: Harbour is just a namespace. This is a Jolla convention and doesn't harm the desktop version. If we get in trouble with this in the future, we can rework the naming...

**Comment by LNJ2** [comment]
I will merge it. (JBB and I have approved with :+1:)

**Pull request merged**